### DISCUSSION TITLE
R Markdown compatibility

### DISCUSSION THREAD

**Discussion by danielvartan** [quarto-dev/quarto-cli#4026] [Q&A] [answered]
I'm having difficulties trying to move from R Markdown to Quarto when rendering GitHub Markdown documents. I can't see how to set up the global variables `fig.path` and  `out.width`  like I would with Rmd:

```{r, include = FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  comment = "#>",
  fig.path = "docs/source/_static/README-",
  out.width = "100%"
)
```

Using Quarto, `out.width` or `fig-width` don't allow strings as values (like `"100%"`). It's impossible to set it like a percentage.

There is no global variable like `fig.path`.

Is there way to do that?

**Comment by cderv** [comment, marked as answer]
> I can't see how to set up the global variables fig.path and out.width like I would with Rmd

First thing is that you can still use the same setup chunk as you have in R Markdown. It should just work without any transformation. Knitr option can be set this way. 

<details>
<summary>Example</summary>

````markdown
---
format: gfm
---

```{r setup, include = FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  comment = "#>",
  fig.path = "docs/source/_static/README-",
  out.width = "100%"
)
```

# Readme

```{r}
1 + 1
```

A figure

```{r}
plot(cars)
```

````



</details>

Regarding a more Quarto way for this, you can set global knitr option in YAML now - see https://quarto.org/docs/computations/execution-options.html#knitr-options

<details>
<summary>Example</summary>

````markdown
---
format: gfm
knitr:
  opts_chunk: 
    collapse: true
    comment: #>
    fig.path: "docs/source/_static/README-"
    out.width: "100%"
---

# Readme

```{r}
1 + 1
```

A figure

```{r}
plot(cars)
```
````

</details>

Can you try the above ? and can you share what is not working for you so that we can have a look ?
